src/lib/mina_generators/parties_generators.ml

```diff
@@ -1343,10 +1326,10 @@ let gen_parties_from ?failure ?(max_other_parties = max_other_parties)
           in
           (* if we use this account again, it will have a Signature authorization *)
           let permissions_auth = Control.Tag.Signature in
```

Comment:
So here, we are resetting the permissions to `Signature` (in party0 we set it to whatever was generated). is this because we are generating an additional signed party in each iteration? if so, could you please add a comment that describes this reason?

Reply:
The original logic of this `gen_parties_with_dynamic_logic` function comes with this logic. I guess it's because it's easier to implement because we could always assume every account comes with signature authorization. There's a comment describe this behavior but there's no comment for the reason behind this.